src/main/java/gyro/google/compute/BackendBucketCdnPolicy.java

```diff
@@ -60,6 +72,169 @@ public void setSignedUrlKeyNames(List<String> signedUrlKeyNames) {
+     */
+    @Updatable
+    public BackendBucketCdnCacheKeyPolicy getCacheKeyPolicy() {
+        return cacheKeyPolicy;
+    }
+
+    public void setCacheKeyPolicy(BackendBucketCdnCacheKeyPolicy cacheKeyPolicy) {
+        this.cacheKeyPolicy = cacheKeyPolicy;
+    }
+
+    /**
+     * Specifies a list of request headers that will be used to bypass cache.
+     *
+     * @subresource gyro.google.compute.BackendBucketCdnBypassCacheOnRequestHeader
+     */
+    @Updatable
+    public List<BackendBucketCdnBypassCacheOnRequestHeader> getBypassCacheOnRequestHeader() {
+        if (bypassCacheOnRequestHeader == null) {
+            bypassCacheOnRequestHeader = new ArrayList<>();
+        }
+
+        return bypassCacheOnRequestHeader;
+    }
+
+    public void setBypassCacheOnRequestHeader(List<BackendBucketCdnBypassCacheOnRequestHeader> bypassCacheOnRequestHeader) {
+        this.bypassCacheOnRequestHeader = bypassCacheOnRequestHeader;
+    }
+
+    /**
+     * Specifies whether to follow redirects from the origin. Defaults to ``false``.
+     */
+    @Updatable
+    public Boolean getNegativeCaching() {
+        if (negativeCaching == null) {
+            negativeCaching = false;
+        }
+
+        return negativeCaching;
+    }
+
+    public void setNegativeCaching(Boolean negativeCaching) {
+        this.negativeCaching = negativeCaching;
+    }
+
+    /**
+     * Specifies the negative caching configuration.
+     *
+     * @subresource gyro.google.compute.BackendBucketCdnNegativeCachingPolicy
+     */
+    @Updatable
+    public List<BackendBucketCdnNegativeCachingPolicy> getNegativeCachingPolicy() {
+        if (negativeCachingPolicy == null) {
+            negativeCachingPolicy = new ArrayList<>();
+        }
+
+        return negativeCachingPolicy;
+    }
+
+    public void setNegativeCachingPolicy(List<BackendBucketCdnNegativeCachingPolicy> negativeCachingPolicy) {
+        this.negativeCachingPolicy = negativeCachingPolicy;
+    }
+
+    /**
+     * Specifies the TTL for cached content served to clients. Defaults to ``0``.
+     */
+    @Updatable
+    public Integer getClientTtl() {
+        if (clientTtl == null) {
+            clientTtl = 0;
+        }
+
+        return clientTtl;
+    }
+
+    public void setClientTtl(Integer clientTtl) {
+        this.clientTtl = clientTtl;
+    }
+
+    /**
+     * Specifies the TTL for cached responses served to clients that do not send a max-age, min-fresh, or s-max-age directive. Defaults to ``0``.
+     */
+    @Updatable
+    public Integer getDefaultTtl() {
+        if (defaultTtl == null) {
+            defaultTtl = 0;
+        }
+
+        return defaultTtl;
+    }
+
+    public void setDefaultTtl(Integer defaultTtl) {
+        this.defaultTtl = defaultTtl;
+    }
+
+    /**
+     * Specifies the maximum allowed TTL for cached content served to clients. Defaults to ``0``.
+     */
+    @Updatable
+    public Integer getMaxTtl() {
+        if (maxTtl == null) {
+            maxTtl = 0;
```

Comment:
This cannot be set to the default 0. Even if I don't specify a value in the configs, get the following error if `cacheMode` is set to `FORCE_CACHE_ALL`.
I think we can remove the defaults from the other `ttl` values too, and not set them in the create/update if the value is not set in the configs.

```
Invalid value for field 'resource.cdnPolicy.cacheMode': 'FORCE_CACHE_ALL'. max_ttl must be specified with CACHE_ALL_STATIC cache_mode only.
```